openrouteservice/src/main/java/org/heigit/ors/fastisochrones/partitioning/InertialFlow.java

```diff
@@ -0,0 +1,333 @@
+        //Estimated maximum iterations
+        int mincutScore = ghGraph.getBaseGraph().getAllEdges().length();
+        double sizeFactor = ((double) projections.get(Projection.LINE_M00).size()) / ghGraph.getBaseGraph().getNodes();
+        mincutScore = Math.max((int) Math.ceil(mincutScore * sizeFactor), 5);
+        BiPartition biPartition = new BiPartition();
+        MaxFlowMinCut maxFlowMinCut = createEdmondsKarp();
+        List<Projection> projOrder = projector.calculateProjectionOrder(projections);
+        //>> Loop through Projections and project each Node
+        int i = 0;
+        for (Projection proj : projOrder) {
+            //Try only best projections
+            if (i == CONSIDERED_PROJECTIONS)
+                break;
+            //>> sort projected Nodes
+            maxFlowMinCut.setOrderedNodes(projections.get(proj));
+            maxFlowMinCut.setNodeOrder();
+            maxFlowMinCut.setMaxFlowLimit(mincutScore);
+            maxFlowMinCut.reset();
+            int cutScore = maxFlowMinCut.getMaxFlow();
+            if (cutScore < mincutScore) {
+                mincutScore = cutScore;
+                biPartition = maxFlowMinCut.calcNodePartition();
+            }
+            i++;
+        }
+        return biPartition;
+    }
+
+    /**
+     * Save multiple cells (from separated disconnected) by iteratively assigning bit shifted cellIds
+     *
+     * @param cells     set of cells to be stored
+     * @param newCellId original cell ids
+     */
+    private void saveMultiCells(Set<IntHashSet> cells, int newCellId) {
+        // Store first cell with given cellId. If there are more subcells to be stored, create subcellId.
+        Iterator<IntHashSet> iterator = cells.iterator();
+        IntHashSet cell = iterator.next();
+        for (IntCursor node : cell) {
+            nodeToCellArr[node.value] = newCellId;
+        }
+        newCellId = newCellId << 1;
+
+        while (iterator.hasNext()) {
+            cell = iterator.next();
+            for (IntCursor node : cell) {
+                nodeToCellArr[node.value] = newCellId;
+            }
+            if (iterator.hasNext()) {
+                cell = iterator.next();
+                for (IntCursor node : cell) {
+                    nodeToCellArr[node.value] = newCellId | 1;
+                }
+            }
+            newCellId = newCellId << 1;
+        }
+    }
+
+    /**
+     * Recursively invoke InertialFlow for partitioning areas of the graph further.
+     * Either invoke a further partition on both partitions or just one.
+     *
+     * @param invokeNext            which partitions to further divide
+     * @param nodesPartition0       size of partition 0
+     * @param nodesPartition1       size of partition 1
+     * @param biPartitionProjection reference to projection
+     */
+    private void recursion(boolean[] invokeNext, int nodesPartition0, int nodesPartition1, BiPartitionProjection biPartitionProjection) {
```

Comment:
This method could probably be simplified by substituting
```
if (invokeNext[0] && invokeNext[1]) { }
else if (invokeNext[0]) { }
else if (invokeNext[1]) { }
```
by
```
if (invokeNext[0]) { }
if (invokeNext[1]) { }
```

Reply:
Done